src/unity/python/turicreate/visualization/plot.py

```diff
@@ -126,33 +127,44 @@ def save(self, filepath, include_data=True):
         >>> plt.save('vega_spec.json', False)
 
         """
-        spec = _json.loads(self.__proxy__.get('call_function', {'__function_name__': 'get_spec'}))
-
-        if(include_data):
-            data = _json.loads(self.__proxy__.get('call_function', {'__function_name__': 'get_data'}))["data_spec"]
-
-            for x in range(0, len(spec["vega_spec"]["data"])):
-                if(spec["vega_spec"]["data"][x]["name"] == "source_2"):
-                    spec["vega_spec"]["data"][x] = data
-                    break;
+        if type(filepath) != str:
+            raise ValueError("File path provided is not a string")
+
+        if filepath.endswith(".json"):
+            # save as vega json
+            spec = self._get_vega(include_data = include_data)
+            with open(filepath, 'w') as fp:
+                _json.dump(spec, fp)
+        elif filepath.endswith(".png") or filepath.endswith(".svg"):
+            # save as png/svg, but json first
+            spec = self._get_vega(include_data = True)
+            extension = filepath[-3:]
+            with open("temp_file.vg.json", 'w') as fp:
+                _json.dump(spec, fp)
+            dirname = os.path.dirname(__file__)
```

Comment:
I don't think it's safe to assume that the directory where turicreate is installed will be writable. 

I think you should use the `tempfile` module:
https://docs.python.org/2/library/tempfile.html